src/x11.rs

```diff
@@ -190,6 +333,33 @@ impl ShmSegment {
             Ok(Self { id, ptr, size })
         }
     }
+
+    /// Copy data into this shared memory segment.
+    ///
+    /// # Safety
+    ///
+    /// This function assumes that the size of `self`'s buffer is larger than or equal to `data.len()`.
```

Comment:
Would it hurt much to make this safe and add an `assert!` or something like that? That extra check should be cheap compared to the `memcpy` that follows.